pkg/kubectl/cmd/delete/delete.go

```diff
@@ -149,6 +150,8 @@ func (o *DeleteOptions) Complete(f cmdutil.Factory, args []string, cmd *cobra.Co
 		return err
 	}
 
+	o.WarnClusterScope = enforceNamespace
```

Comment:
`o.WarnClusterScope = enforceNamespace && !o.DeleteAllNamespaces`?

Reply:
If -n / --namespace is specified, enforceNamespace  would be true.

If --all-namespace is specified, I prefer not to show warning message.

Reply:
Even "--alll" is not specified, warning message shoule be thrown when the user deletes a cluster-scoped resource with -n option

Reply:
Correct, that's what that does